modin/core/dataframe/pandas/exchange/dataframe_protocol/column.py

```diff
@@ -0,0 +1,455 @@
+            "f": DTypeKind.FLOAT,
+            "b": DTypeKind.BOOL,
+        }
+        kind = _np_kinds.get(dtype.kind, None)
+        if kind is None:
+            raise NotImplementedError(
+                f"Data type {dtype} not supported by the dataframe exchange protocol"
+            )
+        return (
+            kind,
+            dtype.itemsize * 8,
+            pandas_dtype_to_arrow_c(dtype),
+            dtype.byteorder,
+        )
+
+    @property
+    def describe_categorical(self) -> Dict[str, Any]:
+        if not self.dtype[0] == DTypeKind.CATEGORICAL:
+            raise RuntimeError(
+                "`describe_categorical only works on a column with "
+                + "categorical dtype!"
+            )
+
+        pandas_series = self._col.to_pandas().squeeze(axis=1)
+        return {
+            "is_ordered": pandas_series.cat.ordered,
+            "is_dictionary": True,
+            "mapping": dict(zip(pandas_series.cat.codes, pandas_series.cat.categories)),
+        }
+
+    @property
+    def describe_null(self) -> Tuple[int, Any]:
+        kind = self.dtype[0]
+        value = None
+        if kind == DTypeKind.FLOAT:
+            null = ColumnNullType.USE_NAN
+        elif kind == DTypeKind.DATETIME:
+            null = ColumnNullType.USE_NAN
+        elif kind in (DTypeKind.INT, DTypeKind.UINT, DTypeKind.BOOL):
+            null = ColumnNullType.NON_NULLABLE
+        elif kind == DTypeKind.CATEGORICAL:
+            # Null values for categoricals are stored as `-1` sentinel values
+            # in the category date (e.g., `col.values.codes` is int8 np.ndarray)
+            null = ColumnNullType.USE_SENTINEL
+            value = -1
+        elif kind == DTypeKind.STRING:
+            null = ColumnNullType.USE_BYTEMASK
+            value = (
+                0  # follow Arrow in using 1 as valid value and 0 for missing/null value
+            )
+        else:
+            raise NotImplementedError(f"Data type {kind} not yet supported")
+
+        return null, value
+
+    _null_count_cache = None
+
+    # TODO: since python 3.9:
+    # @cached_property
+    @property
+    def null_count(self) -> int:
+        if self._null_count_cache is not None:
+            return self._null_count_cache
+
+        def map_func(df):
+            return df.isna()
+
+        def reduce_func(df):
+            return pandas.DataFrame(df.sum())
+
+        intermediate_df = self._col.tree_reduce(0, map_func, reduce_func)
+        # Set ``pandas.RangeIndex(1)`` to index and column labels because
+        # 1) We internally use '__reduced__' for labels of a reduced axis
+        # 2) The return value of `reduce_func` is a pandas DataFrame with
+        # index and column labels set to ``pandas.RangeIndex(1)``
+        # 3) We further use `to_pandas().squeeze()` to get an integer value of the null count
+        intermediate_df.index = pandas.RangeIndex(1)
+        intermediate_df.columns = pandas.RangeIndex(1)
```

Comment:
The answer is in the comments above😄 We can't just use `to_pandas().squeeze()` because we get mismatching internal and external indices for both axes in that case. Expanded the last line of the comments to be even clearer.